lib/instruction-parser.ts

```diff
@@ -0,0 +1,45 @@
+export { getPackagesFromRunInstructions, DockerFilePackages };
+
+interface DockerFilePackages {
+  [packageName: string]: {
+    instruction: string;
+  };
+}
+
+// tslint:disable-next-line:max-line-length
+const installRegex = /\s*(rpm\s+-i|rpm\s+--install|apk\s+add|apt\s+install|apt-get\s+install)\s+(-\w\s+)?/;
+
+/*
+ * This is fairly ugly because a single RUN could contain multiple install
+ * commands, which in turn may install multiple packages, so we've got a
+ * 3-level nested array (RUN instruction[] -> install[] -> package[])
+ *
+ * We also need to account for the multiple ways to split commands, and
+ * arbitrary whitespace
+ */
+function getPackagesFromRunInstructions(
+  runInstructions: string[],
+): DockerFilePackages {
+  return runInstructions.reduce((dockerfilePackages, instruction) => {
+    // A single RUN may include multiple commands, so remove "RUN " and split
+    const commands = instruction.slice(4).split(/\s?(;|&&)\s?/);
```

Comment:
what's 4? the len of 'RUN '?
Maybe create a const and use length?

Reply:
Good point, will do